packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dedicatedCloud.service.js

```diff
@@ -536,6 +536,53 @@ class DedicatedCloudService {
     });
   }
 
+  /* ------- IAM -------*/
+
+  getIamStatus(serviceName) {
+    return this.$http
+      .get(`/dedicatedCloud/${serviceName}/iam`)
+      .then(({ data }) => data);
+  }
+
+  isIamTogglable(serviceName, state, _returnAsBoolean = true) {
+    const returnAsBoolean = _returnAsBoolean;
+
+    if (includes(['disabling', 'enabling'], state)) {
+      // while enabling/disabling activation button not show, no need for a message (the tooltip)
+      return { toggable: false };
+    }
+
+    const endpoint = `canBe${state === 'enabled' ? 'Disabled' : 'Enabled'}`;
+
+    return this.$http
+      .get(`/dedicatedCloud/${serviceName}/iam/${endpoint}`)
+      .then(({ data }) => {
+        // API return a funny response, assume it is true if the response was a 200
+        return returnAsBoolean ? { toggable: true } : data;
+      })
+      .catch((err) => ({ toggable: false, error: err }));
```

Comment:
**nitpick:**  Stick to normal `.then().catch()` to handle the error. Avoid returning customised data/error from the service to make the function reusuable
```suggestion
      
```

Reply:
Sure, will patch that